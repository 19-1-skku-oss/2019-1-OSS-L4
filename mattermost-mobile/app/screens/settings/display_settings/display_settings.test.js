// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';
import SettingsItem from 'app/screens/settings/settings_item';
import Preferences from 'mattermost-redux/constants/preferences';

import DisplaySettings from './display_settings';

jest.mock('react-intl');

describe('DisplaySettings', () => {
    const baseProps = {
        theme: Preferences.THEMES.default,
        enableTheme: false,
        enableTimezone: false,
        navigator: {
            push: jest.fn(),
            setOnNavigatorEvent: jest.fn(),
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <DisplaySettings {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
        expect(wrapper.find(SettingsItem).length).toBe(1);
        wrapper.setProps({enableTheme: true});
        expect(wrapper.find(SettingsItem).length).toBe(2);
        wrapper.setProps({enableTimezone: true});
        expect(wrapper.find(SettingsItem).length).toBe(3);
    });
});
