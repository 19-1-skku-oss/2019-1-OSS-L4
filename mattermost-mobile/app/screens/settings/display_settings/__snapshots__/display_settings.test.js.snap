// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplaySettings should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <View
    style={
      Object {
        "backgroundColor": "rgba(61,60,64,0.06)",
        "flex": 1,
        "paddingTop": 35,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(61,60,64,0.1)",
          "height": 1,
        }
      }
    />
    <SettingsItem
      defaultMessage="Clock Display"
      i18nId="mobile.advanced_settings.clockDisplay"
      iconName="ios-time"
      iconType="ion"
      isDestructor={false}
      onPress={[Function]}
      separator={false}
      showArrow={false}
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(61,60,64,0.1)",
          "height": 1,
        }
      }
    />
  </View>
</View>
`;
